File: 2/linkedlist.cpp
class Node {
public:
    Node next;
    Node previous;
    int data;

    Node(int data) {
        this.data = data;
    }

    void append(int data) {
        Node end = Node(data);
        Node n = this;
        while(n.next != NULL) {
            n = n.next;
        }

        n.next = end;
    }
};

File: linkedlist.py
class Node:
    def __init__(self, data):
        self.data = data

    def append(self, data):
        end = Node(data)
        n = self
        while n.next is not None:
            n = n.next

        n.next = end


def init(*nums):
    head = Node(nums[0])
    for i in range(1, len(nums)):
        head.append(nums[i])
